Unity-C#/README.md

```diff
@@ -0,0 +1,263 @@
+```
+
+
+#### Events
+```csharp
+public ChangedNameHandler OnNameChanged
+```
+
+
+#### Specific Naming Conventions
+
+- The "is" prefix SHOULD be used for boolean variables and methods. Alternatives include "has", "can" and "should".
+
+- The terms "initialize" or "init" CAN be used where an object or a concept is established.
+
+- Plural form MUST be used to name collections.
+
+- Iterator variables SHOULD be called "i", "j", "k", etc.
+
+- Complement names MUST be used for complement entities. Examples: get/set, add/remove, create/destroy, start/stop, insert/delete, begin/end, etc.
+
+- Negated boolean variable names MUST be avoided:
+```csharp
+	bool isNotError, isNotFound; // WRONG.
+```
+
+- Methods returning an object MAY be named after what they return, and methods returning void after what they do.
+
+
+## Variable declarations
+Declare one variable per type statement, it makes it easier to re-order the lines, and put declarations in the order in which they make sense.
+Ideally, variables should be initialized where they are declared and they must be declared in the smallest scope possible.
+
+
+### Loop
+Only loop control statements MUST be included in the "for" loop construction.
+
+Loop variables should be initialized immediately before the loop; loop variables in a "for" statement may be initialized in the "for" loop construction.
+
+The use of "break" and "continue" is not discouraged.
+
+### Conditionals
+Complex conditional expressions SHOULD be avoided; use temporary boolean variables instead with proper naming.
+
+```csharp
+bool isAuthorized = (user.IsAdmin() || user.IsModerator());
+if (isAuthorized)
+{
+	Console.Log('winning');
+}
+```
+
+The nominal case should be put in the "if" part and the exception in the "else" part of an "if" statement.
+
+Executable statements in conditionals MUST be avoided.
+
+### Miscellaneous
+The use of magic numbers in the code should be avoided; they should be declared using named **constants** instead.
+
+
+## Return statement
+To avoid deep nesting of if-statements, always return a function's value as early as possible.
+
+```csharp
+public bool IsPercentage(int percent) 
+{
+	if (percent < 0)
+	{
+		return false;
+	}
+
+	if (percent > 100)
+	{
+		return false;
+	}
+
+	return true;
+}
+
+```
+
+
+## Comments
+
+Comments are the documentation of your code; therefore **tricky code should not be commented but rewritten**. 
+
+All comments should be written in English and be politically correct, even those that are supposed to be temporary.
+
+Comments should be indented relative to their position in the code, preceding the code in question.
+
+Even if everybody likes commented code, do not comment every single line of code; you will kill the readability of it.
+
+Comments are supposed to explain an algorithm, not repeat it in a different language. This is wrong:
+```csharp
+// increment i by 1
+i += 1;
+```
+
+Try to put comment at the top of a block of code to explain the point of it..
```

Comment:
XD I have no idea either, but i think it would be better to have 
/// <Summary>
/// Some really useful comment that actually describes what this function is all about.
/// </Summary>
GetTextEffectCharacterIndex()
{
// Blah
}